reverse_image_search_module/search_image.py

```diff
@@ -39,21 +58,47 @@ def change_format(data):
         'dimension': data['Dimensions'],
         'description': data['WikiDescription'],
         'tags': data['Tags'],
-        'image_url': data['image_urls']
+        'image_url': data['image_urls'],
+        # 'file_name': file_name
     }
 
 
 def find_image(img):
-    plt.imshow(img)
-    plt.show()
-
-    transposed_img = torch.from_numpy(np.transpose(img, (2, 0, 1)))
-    vector = img2vec.getVectors(transposed_img)
+    # print(img.shape)
+    # plt.imshow(img)
+    # plt.show()
+    #
+    #
+    # transposed_img = torch.from_numpy(np.transpose(img, (2, 0, 1)))
+    # #added
+    # transposed_img = transposed_img.to(torch.float)
+    # #
+    vector = img2vec.getVectors(img)
     vector = np.transpose(vector)
+    # vector = np.transpose(vector)
     vector = vector / np.linalg.norm(vector)
 
     idx, dist = annoy_index.get_nns_by_vector(vector, 1, search_k=-1, include_distances=True)
     idx, dist = idx[0], dist[0]
-    data = change_format(dataset.iloc[idx].to_dict())
-
+    file_n = file_names[idx]
+    # matching_idx = dataset[dataset['file_name'] == file_n].index
+    matching_idx = dataset[dataset['file_name'] == file_n].index
+    data = change_format(dataset.loc[matching_idx].to_dict())
     return idx, dist, data
+# import cv2  # OpenCV for image loading and preprocessing
+#from PIL import Image
+
+#image_path = '../data/img/full/9247d9e1c6308bed661ba721f7f35be1fe1f3b52.jpg'
+#image = Image.open(image_path)
+# feature_vector = img2vec.getVectors(image)
+
+# image_path = '../data/img/full/9247d9e1c6308bed661ba721f7f35be1fe1f3b52.jpg'
+# img = cv2.imread(image_path)
+# print(img.shape)
+# img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+# print(feature_vector.shape)
+#idx, dist, data = find_image(image)
+
+#print("Index:", idx)
+#print("Distance:", dist)
```

Comment:
create a test

Reply:
@michaelvin1322 did you wanted to create tests for this?

Reply:
@mjason98 Alex told that you are writing the test
You can use this piece of code.